src/hardware/sblaster.cpp

```diff
@@ -1011,21 +1011,67 @@ static std::vector<AudioFrame>& maybe_silence(const T* samples,
 	return frames;
 }
 
-static uint32_t read_dma_8bit(const uint32_t bytes_to_read, const uint32_t i = 0)
+static uint32_t read_dma_8bit(const uint32_t bytes_to_read, const uint32_t buffer_index = 0)
 {
-	const auto bytes_read = sb.dma.chan->Read(bytes_to_read, sb.dma.buf.b8 + i);
-	assert(bytes_read <= DmaBufSize * sizeof(sb.dma.buf.b8[0]));
+	static_assert(sizeof(sb.dma.buf.b8[0]) == 1);
+
+	if (buffer_index >= DmaBufSize) {
+		// Should never happen as the code is currently written.
+		// Calling code has buffer_index either 0 or 1 to handle a dangling sample from the last read.
```

Comment:
It would be also useful to explain here briefly the problem that "dangling sample" solves (padding stereo sample streams).

Reply:
done